### PR TITLE
Bug 527415 #345 - Fix code when locale is tr, az or lt

### PR DESCRIPTION
**Pull request by ajacob**
The bug from #345 is not the same that was reported in 527415, I corrected both of them.
Changed more code that I should have but tried to be defensive to avoid other similar bugs.

This blog post gives a good explanation of the problem : https://garygregory.wordpress.com/2015/11/03/java-lowercase-conversion-turkey/

PS : Is it mandatory to add ourself to Contributors at the beggining of each file ?

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=527415

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=527415

File: jpa/org.eclipse.persistence.jpa.jpql.test/src/org/eclipse/persistence/jpa/tests/jpql/parser/JPQLExpressionTurkishLocaleTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.eclipse.persistence.jpa.jpql.parser.JPQLExpression;
+import org.eclipse.persistence.jpa.jpql.parser.JPQLGrammar;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class JPQLExpressionTurkishLocaleTest {
+
+    @Test
+    public void testJPQLExpressionWithTurkishLocale() {
+        Locale.setDefault(new Locale("tr", "TR"));
```
Comment (lukasj) [inline jpa/org.eclipse.persistence.jpa.jpql.test/src/org/eclipse/persistence/jpa/tests/jpql/parser/JPQLExpressionTurkishLocaleTest.java]: test should reset this to default when it's done so other tests, if run in the same VM are not affected

**Comment by andreysubbotin** [comment]
Any updates?

**Pull request merged**